salt/modules/zypperpkg.py

```diff
@@ -2528,7 +2528,7 @@ def download(*packages, **kwargs):
     )
 
 
-def list_downloaded(root=None):
+def list_downloaded(root=None, **kwargs):
```

Comment:
We don't use kwargs in list_downloaded. 
I believe this would be a better fix.
OLD
file "/usr/lib/python3.7/site-packages/salt/states/pkg.py", line 268, in _find_download_targets
                  cur_pkgs = __salt__['pkg.list_downloaded'](**kwargs)
NEW
file "/usr/lib/python3.7/site-packages/salt/states/pkg.py", line 268, in _find_download_targets
                  cur_pkgs = __salt__['pkg.list_downloaded'](kwargs.get('root'))

Reply:
@cmcmarrow I like this idea, but in that case I would also change the signature here: https://github.com/saltstack/salt/blob/develop/salt/modules/yumpkg.py#L991

Also there are other places that require the change in pkg.py:

https://github.com/saltstack/salt/blob/develop/salt/states/pkg.py#L2166

Reply:
Hey @cmcmarrow, I understand your point but I've just added `kwargs` as an accepted argument to `list_downloaded` since that's the way most of the Zypper module (and also yumpkg) is implemented. There are many methods accepting `kwargs` even if they are not used at the moment:

```
 732 def version_cmp(ver1, ver2, ignore_epoch=False, **kwargs):                                                                                                              
1039 def get_repo(repo, root=None, **kwargs):  # pylint: disable=unused-argument                                                                                             
1055 def list_repos(root=None, **kwargs):
1556 def upgrade(refresh=True, dryrun=False, dist_upgrade=False, fromrepo=None, novendorchange=False, skip_verify=False, no_recommends=False, root=None, **kwargs):  # pylint: disable=unused-argument
1753 def remove(name=None, pkgs=None, root=None, **kwargs):  # pylint: disable=unused-argument                                                                               
1800 def purge(name=None, pkgs=None, root=None, **kwargs):  # pylint: disable=unused-argument
1960 def remove_lock(packages, root=None, **kwargs):  # pylint: disable=unused-argument
2044 def add_lock(packages, root=None, **kwargs):  # pylint: disable=unused-argument
2647 def list_patches(refresh=False, root=None, **kwargs):
2673 def list_installed_patches(root=None, **kwargs):
```

One of the reason for this might be to prevent changing the method signature (also for APIs) in case that there is a new argument that wants to be introduced to one of those method.

Reply:
okay that's fair